pkg/vmprovider/providers/vsphere2/vmprovider_vm.go

```diff
@@ -951,9 +936,7 @@ func (vs *vSphereVMProvider) vmCreateGenConfigSpecExtraConfig(
 		ecMap[k] = renderTemplateFn(k, v)
 	}
 
-	hasPassthroughDevices := len(util.SelectVirtualPCIPassthrough(util.DevicesFromConfigSpec(createArgs.ConfigSpec))) > 0
```

Comment:
This was slightly different than what we had on the config spec side - this would only check the type for passthrough, not the backing for vGPU/DDPIO. I don't think we intentionally made this different, and I think just checking the type is what was intended.